Line and Plane versions of bisector fails to compile

_Please use the following template to help us solving your issue._

## Issue Details

Compilation fails when trying to instantiate `Line_2` and `Plane_3` versions of `CGAL::bisector()` in CGAL 4.14

Compiler warnings for `Line_2` version given below:

```
In file included from bisector_test.cpp:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Exact_predicates_exact_constructions_kernel.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Simple_cartesian.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/Cartesian_base.h:37:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/Line_2.h:31:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/predicates/kernel_ftC2.h:33:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/constructions/kernel_ftC2.h:258:6: error: no viable conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>'
  FT n1 = CGAL_NTS sqrt(CGAL_NTS square(pa) + CGAL_NTS square(pb));
     ^    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/function_objects.h:1953:7: note: in instantiation of function template specialization 'CGAL::bisector_of_linesC2<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> >' requested here
      bisector_of_linesC2(p.a(), p.b(), p.c(),
      ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Lazy.h:1698:33: note: in instantiation of member function 'CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> > >::operator()' requested here
  BOOST_PP_REPEAT_FROM_TO(1, 9, CGAL_CONSTRUCTION_OPERATOR, _)
                                ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/repetition/repeat_from_to.hpp:34:34: note: expanded from macro 'BOOST_PP_REPEAT_FROM_TO'
# define BOOST_PP_REPEAT_FROM_TO BOOST_PP_CAT(BOOST_PP_REPEAT_FROM_TO_, BOOST_PP_AUTO_REC(BOOST_PP_REPEAT_P, 4))
                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/cat.hpp:22:32: note: expanded from macro 'BOOST_PP_CAT'
#    define BOOST_PP_CAT(a, b) BOOST_PP_CAT_I(a, b)
                               ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro 'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:16:1: note: expanded from here
BOOST_PP_REPEAT_FROM_TO_1
^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Kernel/global_functions_internal_2.h:182:10: note: in instantiation of function template specialization 'CGAL::Lazy_construction<CGAL::Epeck, CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<CGAL::Interval_nt<false> > >, CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> > >, CGAL::Default, true>::operator()<CGAL::Line_2<CGAL::Epeck>, CGAL::Line_2<CGAL::Epeck> >' requested here
  return k.construct_bisector_2_object()(l1, l2);
         ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Kernel/global_functions_2.h:185:20: note: in instantiation of function template specialization 'CGAL::internal::bisector<CGAL::Epeck>' requested here
  return internal::bisector(l1, l2, K());
                   ^
bisector_test.cpp:8:16: note: in instantiation of function template specialization 'CGAL::bisector<CGAL::Epeck>' requested here
  return CGAL::bisector(line1, line2);
               ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:56:41: note: candidate constructor not viable: no known conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'const boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> &' for 1st argument
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const number& e) BOOST_MP_NOEXCEPT_IF(noexcept(Backend(std::declval<Backend const&>()))) : m_backend(e.m_backend) {}
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:317:41: note: candidate constructor not viable: no known conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> &&' for 1st argument
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(number&& r)
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:58:50: note: candidate template ignored: substitution failure [with V = CGAL::Null_tag]: no type named 'type' in 'boost::enable_if_c<false, void>'
   BOOST_MP_FORCEINLINE BOOST_MP_CXX14_CONSTEXPR number(const V& v, typename boost::enable_if_c<
                                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:68:41: note: candidate template ignored: substitution failure [with V = CGAL::Null_tag]: no type named 'type' in 'boost::enable_if_c<false, void>'
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const V& v, typename boost::enable_if_c<
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:103:41: note: candidate template ignored: could not match 'number<boost::multiprecision::backends::gmp_rational, ExpressionTemplates>' against 'CGAL::Null_tag'
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const number<Backend, ET>& val)
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:107:50: note: candidate template ignored: could not match 'number<type-parameter-0-0, ExpressionTemplates>' against 'CGAL::Null_tag'
   BOOST_MP_FORCEINLINE BOOST_MP_CXX14_CONSTEXPR number(const number<Other, ET>& val,
                                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:299:29: note: candidate template ignored: could not match 'expression<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3, type-parameter-0-4>' against 'CGAL::Null_tag'
   BOOST_MP_CXX14_CONSTEXPR number(const detail::expression<tag, Arg1, Arg2, Arg3, Arg4>& e, typename boost::enable_if_c<is_convertible<typename detail::expression<tag, Arg1, Arg2, Arg3, Arg4>::result_type, self_type>::value>::type* = 0)
                            ^
In file included from bisector_test.cpp:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Exact_predicates_exact_constructions_kernel.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Simple_cartesian.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/Cartesian_base.h:37:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/Line_2.h:31:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/predicates/kernel_ftC2.h:33:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/constructions/kernel_ftC2.h:259:6: error: no viable conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>'
  FT n2 = CGAL_NTS sqrt(CGAL_NTS square(qa) + CGAL_NTS square(qb));
     ^    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:56:41: note: candidate constructor not viable: no known conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'const boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> &' for 1st argument
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const number& e) BOOST_MP_NOEXCEPT_IF(noexcept(Backend(std::declval<Backend const&>()))) : m_backend(e.m_backend) {}
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:317:41: note: candidate constructor not viable: no known conversion from 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt::result_type' (aka 'CGAL::Null_tag') to 'boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> &&' for 1st argument
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(number&& r)
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:58:50: note: candidate template ignored: substitution failure [with V = CGAL::Null_tag]: no type named 'type' in 'boost::enable_if_c<false, void>'
   BOOST_MP_FORCEINLINE BOOST_MP_CXX14_CONSTEXPR number(const V& v, typename boost::enable_if_c<
                                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:68:41: note: candidate template ignored: substitution failure [with V = CGAL::Null_tag]: no type named 'type' in 'boost::enable_if_c<false, void>'
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const V& v, typename boost::enable_if_c<
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:103:41: note: candidate template ignored: could not match 'number<boost::multiprecision::backends::gmp_rational, ExpressionTemplates>' against 'CGAL::Null_tag'
   BOOST_MP_FORCEINLINE BOOST_CONSTEXPR number(const number<Backend, ET>& val)
                                        ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:107:50: note: candidate template ignored: could not match 'number<type-parameter-0-0, ExpressionTemplates>' against 'CGAL::Null_tag'
   BOOST_MP_FORCEINLINE BOOST_MP_CXX14_CONSTEXPR number(const number<Other, ET>& val,
                                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/multiprecision/number.hpp:299:29: note: candidate template ignored: could not match 'expression<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2, type-parameter-0-3, type-parameter-0-4>' against 'CGAL::Null_tag'
   BOOST_MP_CXX14_CONSTEXPR number(const detail::expression<tag, Arg1, Arg2, Arg3, Arg4>& e, typename boost::enable_if_c<is_convertible<typename detail::expression<tag, Arg1, Arg2, Arg3, Arg4>::result_type, self_type>::value>::type* = 0)
                            ^
In file included from bisector_test.cpp:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Exact_predicates_exact_constructions_kernel.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Simple_cartesian.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/Cartesian_base.h:29:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/basic.h:45:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/number_type_basic.h:60:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/double.h:31:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/number_utils.h:104:12: error: type 'typename Algebraic_structure_traits<expression<add_immediates, number<gmp_rational, boost::multiprecision::et_on>, number<gmp_rational, boost::multiprecision::et_on>, void, void> >::Sqrt' (aka 'CGAL::Null_functor') does not provide a call operator
    return sqrt( x );
           ^~~~
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/constructions/kernel_ftC2.h:258:20: note: in instantiation of function template specialization 'CGAL::sqrt<boost::multiprecision::detail::expression<boost::multiprecision::detail::add_immediates, boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>, boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>, void, void> >' requested here
  FT n1 = CGAL_NTS sqrt(CGAL_NTS square(pa) + CGAL_NTS square(pb));
                   ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Cartesian/function_objects.h:1953:7: note: in instantiation of function template specialization 'CGAL::bisector_of_linesC2<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> >' requested here
      bisector_of_linesC2(p.a(), p.b(), p.c(),
      ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Lazy.h:1698:33: note: in instantiation of member function 'CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> > >::operator()' requested here
  BOOST_PP_REPEAT_FROM_TO(1, 9, CGAL_CONSTRUCTION_OPERATOR, _)
                                ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/repetition/repeat_from_to.hpp:34:34: note: expanded from macro 'BOOST_PP_REPEAT_FROM_TO'
# define BOOST_PP_REPEAT_FROM_TO BOOST_PP_CAT(BOOST_PP_REPEAT_FROM_TO_, BOOST_PP_AUTO_REC(BOOST_PP_REPEAT_P, 4))
                                 ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/cat.hpp:22:32: note: expanded from macro 'BOOST_PP_CAT'
#    define BOOST_PP_CAT(a, b) BOOST_PP_CAT_I(a, b)
                               ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro 'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:16:1: note: expanded from here
BOOST_PP_REPEAT_FROM_TO_1
^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Kernel/global_functions_internal_2.h:182:10: note: in instantiation of function template specialization 'CGAL::Lazy_construction<CGAL::Epeck, CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<CGAL::Interval_nt<false> > >, CGAL::CartesianKernelFunctors::Construct_bisector_2<CGAL::Simple_cartesian<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on> > >, CGAL::Default, true>::operator()<CGAL::Line_2<CGAL::Epeck>, CGAL::Line_2<CGAL::Epeck> >' requested here
  return k.construct_bisector_2_object()(l1, l2);
         ^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/cgal4h/include/CGAL/Kernel/global_functions_2.h:185:20: note: in instantiation of function template specialization 'CGAL::internal::bisector<CGAL::Epeck>' requested here
  return internal::bisector(l1, l2, K());
                   ^
bisector_test.cpp:8:16: note: in instantiation of function template specialization 'CGAL::bisector<CGAL::Epeck>' requested here
  return CGAL::bisector(line1, line2);
               ^
```

## Source Code

```c++
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Line_2.h>

typedef CGAL::Exact_predicates_exact_constructions_kernel Kernel;
typedef Kernel::Line_2 Line_2;

Line_2 test_bisector(const Line_2& line1, const Line_2& line2) {
  return CGAL::bisector(line1, line2);
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): macOS
* Compiler: clang
* Release or debug mode: release
* Specific flags used (if any): CGAL_HEADER_ONLY tried with and without CGAL_USE_GMPXX
* CGAL version: 4.14
* Boost version: 1.72.0-3
* Other libraries versions if used (Eigen, TBB, etc.):

--- a/Algebraic_foundations/include/CGAL/number_utils.h
+++ b/Algebraic_foundations/include/CGAL/number_utils.h
@@ -302,13 +302,13 @@ to_interval( const Real_embeddable& x) {
 }
 
 template <typename NT>
-NT approximate_sqrt(const NT& nt, CGAL::Field_tag)
+NT approximate_sqrt(const NT& nt, CGAL::Null_functor)
 {
   return NT(sqrt(CGAL::to_double(nt)));
 }
 
-template <typename NT>
-NT approximate_sqrt(const NT& nt, CGAL::Field_with_sqrt_tag)
+template <typename NT, typename Sqrt>
+NT approximate_sqrt(const NT& nt, Sqrt sqrt)
 {
   return sqrt(nt);
 }
@@ -317,8 +317,8 @@ template <typename NT>
 NT approximate_sqrt(const NT& nt)
 {
   typedef CGAL::Algebraic_structure_traits<NT> AST;
-  typedef typename AST::Algebraic_category Algebraic_category;
-  return approximate_sqrt(nt, Algebraic_category());
+  typedef typename AST::Sqrt Sqrt;
+  return approximate_sqrt(nt, Sqrt());
 }
 
 CGAL_NTS_END_NAMESPACE

--- a/Algebraic_foundations/include/CGAL/number_utils.h
+++ b/Algebraic_foundations/include/CGAL/number_utils.h
@@ -316,6 +316,9 @@ NT approximate_sqrt(const NT& nt, Sqrt sqrt)
 template <typename NT>
 NT approximate_sqrt(const NT& nt)
 {
+  // the initial version of this function was using Algebraic_category
+  // for the dispatch but some ring type (like Gmpz) provides a Sqrt
+  // functor even if not being Field_with_sqrt.
   typedef CGAL::Algebraic_structure_traits<NT> AST;
   typedef typename AST::Sqrt Sqrt;
   return approximate_sqrt(nt, Sqrt());

--- a/Cartesian_kernel/include/CGAL/constructions/kernel_ftC2.h
+++ b/Cartesian_kernel/include/CGAL/constructions/kernel_ftC2.h
@@ -246,8 +246,8 @@ bisector_of_linesC2(const FT &pa, const FT &pb, const FT &pc,
                     FT &a, FT &b, FT &c)
 {
   // We normalize the equations of the 2 lines, and we then add them.
-  FT n1 = CGAL_NTS sqrt(CGAL_NTS square(pa) + CGAL_NTS square(pb));
-  FT n2 = CGAL_NTS sqrt(CGAL_NTS square(qa) + CGAL_NTS square(qb));
+  FT n1 = CGAL_NTS approximate_sqrt( FT(CGAL_NTS square(pa) + CGAL_NTS square(pb)) );
+  FT n2 = CGAL_NTS approximate_sqrt( FT(CGAL_NTS square(qa) + CGAL_NTS square(qb)) );
   a = n2 * pa + n1 * qa;
   b = n2 * pb + n1 * qb;
   c = n2 * pc + n1 * qc;

--- a/Cartesian_kernel/include/CGAL/constructions/kernel_ftC3.h
+++ b/Cartesian_kernel/include/CGAL/constructions/kernel_ftC3.h
@@ -366,10 +366,10 @@ bisector_of_planesC3(const FT &pa, const FT &pb, const FT &pc, const FT &pd,
                      FT &a, FT &b, FT &c, FT &d)
 {
   // We normalize the equations of the 2 planes, and we then add them.
-  FT n1 = CGAL_NTS sqrt(CGAL_NTS square(pa) + CGAL_NTS square(pb) +
-                        CGAL_NTS square(pc));
-  FT n2 = CGAL_NTS sqrt(CGAL_NTS square(qa) + CGAL_NTS square(qb) +
-                        CGAL_NTS square(qc));
+  FT n1 = CGAL_NTS approximate_sqrt( FT(CGAL_NTS square(pa) + CGAL_NTS square(pb) +
+                        CGAL_NTS square(pc)) );
+  FT n2 = CGAL_NTS approximate_sqrt( FT(CGAL_NTS square(qa) + CGAL_NTS square(qb) +
+                        CGAL_NTS square(qc)) );
   a = n2 * pa + n1 * qa;
   b = n2 * pb + n1 * qb;
   c = n2 * pc + n1 * qc;

--- a/Kernel_23/doc/Kernel_23/CGAL/Kernel/global_functions.h
+++ b/Kernel_23/doc/Kernel_23/CGAL/Kernel/global_functions.h
@@ -355,8 +355,10 @@ through the intersection of `l1` and `l2`.
 If `l1` and `l2` are parallel, then the bisector is defined as the line
 which has the same direction as `l1`, and which is at the same distance
 from `l1` and `l2`.
-This function requires that `Kernel::RT` supports the `sqrt()`
-operation.
+If `Kernel::FT` is not a model of `FieldWithSqrt`
+an approximation of the square root will be used in this function,
+impacting the exactness of the result even with a (exact) multiprecision
+number type.
 */
 template <typename Kernel>
 CGAL::Line_2<Kernel> bisector(const CGAL::Line_2<Kernel> &l1,
@@ -379,8 +381,10 @@ passes through the intersection of `h1` and `h2`.
 If `h1` and `h2` are parallel, then the bisector is defined as the
 plane which has the same oriented normal vector as `l1`, and which is at
 the same distance from `h1` and `h2`.
-This function requires that `Kernel::RT` supports the `sqrt()`
-operation.
+If `Kernel::FT` is not a model of `FieldWithSqrt`
+an approximation of the square root will be used in this function,
+impacting the exactness of the result even with a (exact) multiprecision
+number type.
 */
 template <typename Kernel>
 CGAL::Plane_3<Kernel> bisector(const CGAL::Plane_3<Kernel> &h1,

--- a/Kernel_23/doc/Kernel_23/CGAL/Kernel/global_functions.h
+++ b/Kernel_23/doc/Kernel_23/CGAL/Kernel/global_functions.h
@@ -357,7 +357,7 @@ which has the same direction as `l1`, and which is at the same distance
 from `l1` and `l2`.
 If `Kernel::FT` is not a model of `FieldWithSqrt`
 an approximation of the square root will be used in this function,
-impacting the exactness of the result even with a (exact) multiprecision
+impacting the exactness of the result even with an (exact) multiprecision
 number type.
 */
 template <typename Kernel>
@@ -383,7 +383,7 @@ plane which has the same oriented normal vector as `l1`, and which is at
 the same distance from `h1` and `h2`.
 If `Kernel::FT` is not a model of `FieldWithSqrt`
 an approximation of the square root will be used in this function,
-impacting the exactness of the result even with a (exact) multiprecision
+impacting the exactness of the result even with an (exact) multiprecision
 number type.
 */
 template <typename Kernel>

--- a/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
+++ b/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
@@ -3848,8 +3848,10 @@ public:
     If `l1` and `l2` are parallel, then the bisector is defined as the line
     which has the same direction as `l1`, and which is at the same distance
     from `l1` and `l2`.
-    This function requires that `Kernel::RT` supports the `sqrt()`
-    operation.
+    If `Kernel::FT` is not a model of `FieldWithSqrt`
+    an approximation of the square root will be used in this function,
+    impacting the exactness of the result even with a (exact) multiprecision
+    number type.
   */
   Kernel::Line_2 operator()(const Kernel::Line_2&l1,
                             const Kernel::Line_2&l2);
@@ -3891,8 +3893,10 @@ public:
     If `h1` and `h2` are parallel, then the bisector is defined as the
     plane which has the same oriented normal vector as `h1`, and which is at
     the same distance from `h1` and `h2`.
-    This function requires that `Kernel::RT` supports the `sqrt()`
-    operation.
+    If `Kernel::FT` is not a model of `FieldWithSqrt`
+    an approximation of the square root will be used in this function,
+    impacting the exactness of the result even with a (exact) multiprecision
+    number type.
   */
   Kernel::Plane_3 operator()(const Kernel::Plane_3&h1,
                              const Kernel::Plane_3&h2);

--- a/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
+++ b/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
@@ -3850,7 +3850,7 @@ public:
     from `l1` and `l2`.
     If `Kernel::FT` is not a model of `FieldWithSqrt`
     an approximation of the square root will be used in this function,
-    impacting the exactness of the result even with a (exact) multiprecision
+    impacting the exactness of the result even with an (exact) multiprecision
     number type.
   */
   Kernel::Line_2 operator()(const Kernel::Line_2&l1,
@@ -3895,7 +3895,7 @@ public:
     the same distance from `h1` and `h2`.
     If `Kernel::FT` is not a model of `FieldWithSqrt`
     an approximation of the square root will be used in this function,
-    impacting the exactness of the result even with a (exact) multiprecision
+    impacting the exactness of the result even with an (exact) multiprecision
     number type.
   */
   Kernel::Plane_3 operator()(const Kernel::Plane_3&h1,

--- a/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
+++ b/Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h
@@ -3848,10 +3848,6 @@ public:
     If `l1` and `l2` are parallel, then the bisector is defined as the line
     which has the same direction as `l1`, and which is at the same distance
     from `l1` and `l2`.
-    If `Kernel::FT` is not a model of `FieldWithSqrt`
-    an approximation of the square root will be used in this function,
-    impacting the exactness of the result even with an (exact) multiprecision
-    number type.
   */
   Kernel::Line_2 operator()(const Kernel::Line_2&l1,
                             const Kernel::Line_2&l2);
@@ -3893,10 +3889,6 @@ public:
     If `h1` and `h2` are parallel, then the bisector is defined as the
     plane which has the same oriented normal vector as `h1`, and which is at
     the same distance from `h1` and `h2`.
-    If `Kernel::FT` is not a model of `FieldWithSqrt`
-    an approximation of the square root will be used in this function,
-    impacting the exactness of the result even with an (exact) multiprecision
-    number type.
   */
   Kernel::Plane_3 operator()(const Kernel::Plane_3&h1,
                              const Kernel::Plane_3&h2);

--- a/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_line_2.h
+++ b/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_line_2.h
@@ -53,7 +53,6 @@ _test_fct_line_2(const R& )
  std::cout << "Testing functions Line_2" ;
 
  typedef typename  R::RT       RT;
- typedef typename  R::FT       FT;
  typedef typename  R::Point_2  Point_2;
  typedef typename  R::Line_2   Line_2;
 

--- a/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_line_2.h
+++ b/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_line_2.h
@@ -18,22 +18,9 @@
 #ifndef CGAL__TEST_FCT_LINE_2_H
 #define CGAL__TEST_FCT_LINE_2_H
 
-// Accessory function testing functions that require sqrt().
-// Doesn't instantiate anything if RT doesn't support sqrt().
 template <class R>
 bool
-_test_fct_line_sqrt_2(const R&, CGAL::Tag_false)
-{
-//  bool UNTESTED_STUFF_BECAUSE_SQRT_IS_NOT_SUPPORTED;
-  std::cout << std::endl
-            << "NOTE : FT doesn't support sqrt(),"
-               " hence some functions are not tested." << std::endl;
-  return true;
-}
-
-template <class R>
-bool
-_test_fct_line_sqrt_2(const R&, CGAL::Tag_true)
+_test_fct_line_sqrt_2(const R&)
 {
  typedef typename  R::Point_2  Point_2;
  typedef typename  R::Line_2   Line_2;
@@ -178,13 +165,9 @@ _test_fct_line_2(const R& )
  assert(bl1.oriented_side(p2) == CGAL::ON_POSITIVE_SIDE);
  assert( CGAL::parallel(bl1, bl2) );
 
- // More tests, that require sqrt().
- {
-     typedef ::CGAL::Algebraic_structure_traits<FT> AST;
-     static const bool has_sqrt =
-         ! ::boost::is_same< ::CGAL::Null_functor, typename AST::Sqrt >::value;
-     _test_fct_line_sqrt_2(R(), ::CGAL::Boolean_tag<has_sqrt>());
- }
+ // More tests, that require sqrt() or use approx.
+  _test_fct_line_sqrt_2(R());
+
  std::cout << "done" << std::endl;
  return true;
 }

--- a/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_plane_3.h
+++ b/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_plane_3.h
@@ -17,22 +17,9 @@
 #ifndef CGAL__TEST_FCT_PLANE_3_H
 #define CGAL__TEST_FCT_PLANE_3_H
 
-// Accessory function testing functions that require sqrt().
-// Doesn't instantiate anything if RT doesn't support sqrt().
 template <class R>
 bool
-_test_fct_plane_sqrt_3(const R&, CGAL::Tag_false)
-{
-//  bool UNTESTED_STUFF_BECAUSE_SQRT_IS_NOT_SUPPORTED;
-  std::cout << std::endl
-            << "NOTE : FT doesn't support sqrt(),"
-               " hence some functions are not tested." << std::endl;
-  return true;
-}
-
-template <class R>
-bool
-_test_fct_plane_sqrt_3(const R&, CGAL::Tag_true)
+_test_fct_plane_sqrt_3(const R&)
 {
  typedef typename  R::Point_3  Point_3;
  typedef typename  R::Plane_3  Plane_3;
@@ -97,11 +84,8 @@ _test_fct_plane_3(const R& )
  assert(   CGAL::parallel(h1, h2) );
  assert( ! CGAL::parallel(h1, h5) );
 
- // More tests, that require sqrt().
- typedef ::CGAL::Algebraic_structure_traits<FT> AST;
- static const bool has_sqrt =
-     ! ::boost::is_same< ::CGAL::Null_functor, typename AST::Sqrt >::value;
- _test_fct_plane_sqrt_3(R(), ::CGAL::Boolean_tag<has_sqrt>());
+ // More tests, that require sqrt() or use approx.
+ _test_fct_plane_sqrt_3(R());
 
  std::cout << "done" << std::endl;
  return true;

--- a/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_plane_3.h
+++ b/Kernel_23/test/Kernel_23/include/CGAL/_test_fct_plane_3.h
@@ -53,7 +53,6 @@ _test_fct_plane_3(const R& )
  std::cout << "Testing functions Plane_3" ;
 
  typedef typename  R::RT       RT;
- typedef typename  R::FT       FT;
  typedef typename  R::Point_3  Point_3;
  typedef typename  R::Plane_3  Plane_3;
 

